src/AppBundle/Entity/User.php

```diff
@@ -163,4 +163,28 @@ public function eraseCredentials()
         // if you had a plainPassword property, you'd nullify it here
         // $this->plainPassword = null;
     }
+
+    /** @see \Serializable::serialize() */
+    public function serialize()
+    {
+        return serialize([
+            $this->id,
+            $this->username,
+            $this->password,
```

Comment:
This explanation is not very good 😕   Saying *"maybe you need this ... maybe you don't"* without explaining anything else doesn't look good. And then, saying *"you can only serialize "id" ... but maybe it doesn't work because Symfony wants more"* is also confusing.

As usual, this is caused by the massive internal complexity of the security component ... so I don't know what can we do about this.